sdks/java/core/src/main/java/org/apache/beam/sdk/util/NullCredential.java

```diff
@@ -0,0 +1,53 @@
+/*
```

Comment:
1. Users will see the same error message: NULL_CREDENTIAL_REASON.

2. There will be few lines differences about where the exception is thrown.
We typically do:
someClient = Transport.newSomeClient(); // previous throws in here
// few lines between
someClient.list/insert(); // Now throws in here.

I think this changes should apply to GCP data services: BigQuery, PubSub, Storage, but not CloudResourceManager. Updated to get around this.